internal/painter/image.go

```diff
@@ -49,17 +49,27 @@ func PaintImage(img *canvas.Image, c fyne.Canvas, width, height int) image.Image
 					fyne.LogError("SVG Load error:", err)
 					return nil
 				}
-				icon.SetTarget(0, 0, float64(width), float64(height))
 
 				w, h := int(icon.ViewBox.W), int(icon.ViewBox.H)
+				aspect := float32(w) / float32(h)
+				viewAspect := float32(width) / float32(height)
+
+				ww, hh := width, height
+				if viewAspect > aspects[img.Resource] {
+					ww = int(float32(height) * aspect)
+				} else if viewAspect < aspects[img.Resource] {
+					hh = int(float32(width) / aspect)
+				}
+
+				icon.SetTarget(0, 0, float64(ww), float64(hh))
```

Comment:
There are three pairs of variables for width and height in here. It isn’t obvious what the difference between `width`, `w` and `ww` is.
I suggest to rename `w` and `h` to `sourceWidth` and `sourceHeight` (or `srcW` and `srcH`) and the `ww` and `hh` to `targetWidth` and `targetHeight` or `tgtW` and `tgtH`.

Reply:
Or maybe `original` (`orig`) and `texture` (`tex`) instead of `source` and `target` if this fits better.

Reply:
Yeah, I should have fixed this before pushing - thanks